### PR TITLE
default namespace setup

### PR DESCRIPTION
**Pull request by everesio**
## What was changed
Optional job to create a default namespace.

## Why?
Current deployment needs a manual step 

## Checklist

1. Closes 

2. How was this tested:
Tested on k8s with postgresql DB created by zalando postgres operator

3. Any docs updates needed?

### CODE REVIEW

**Review by bergundy — COMMENTED** [review]
Why not let the helm chart register any number of arbitrary namespaces?
It’s more flexible, not called “default” (which may be confused with the name default) and doesn’t need an `enabled` flag.

File: templates/server-job.yaml [hunk 1]
```diff
@@ -301,3 +301,64 @@ spec:
...
+      containers:
+        - name: create-default-namespace
+          image: "{{ $.Values.admintools.image.repository }}:{{ $.Values.admintools.image.tag }}"
+          imagePullPolicy: {{ $.Values.admintools.image.pullPolicy }}
+          command: ['sh', '-c', 'tctl --ns $(DEFAULT_NAMESPACE) namespace describe || tctl --ns $(DEFAULT_NAMESPACE) namespace register --rd $(DEFAULT_NAMESPACE_RETENTION) --desc "Default namespace for Temporal Server."']
+          env:
+            - name: DEFAULT_NAMESPACE
+              value: "{{ .Values.defaultnamespace.setup.name }}"
+            - name: DEFAULT_NAMESPACE_RETENTION
+              value: "{{ .Values.defaultnamespace.setup.retention }}"
```
Comment (bergundy) [inline templates/server-job.yaml:349]: This doesn’t support tls afaict, which is a fairly common requirement when setting up the server.

File: templates/server-job.yaml [hunk 2]
```diff
@@ -301,3 +301,64 @@ spec:
...
+        app.kubernetes.io/component: auto-setup
+        app.kubernetes.io/part-of: {{ .Chart.Name }}
+    spec:
+      {{ include "temporal.serviceAccount" . }}
+      restartPolicy: "OnFailure"
+      containers:
+        - name: create-default-namespace
+          image: "{{ $.Values.admintools.image.repository }}:{{ $.Values.admintools.image.tag }}"
+          imagePullPolicy: {{ $.Values.admintools.image.pullPolicy }}
+          command: ['sh', '-c', 'tctl --ns $(DEFAULT_NAMESPACE) namespace describe || tctl --ns $(DEFAULT_NAMESPACE) namespace register --rd $(DEFAULT_NAMESPACE_RETENTION) --desc "Default namespace for Temporal Server."']
```
Comment (bergundy) [inline templates/server-job.yaml:344]: Can you please use temporal cli instead of the soon to be deprecated tctl?

Comment (bergundy) [reply templates/server-job.yaml:344]: I think this command is missing quotes around the variables

**Review by nikanmog — APPROVED** [review]

File: templates/server-job.yaml [hunk 3]
```diff
@@ -301,3 +301,64 @@ spec:
       {{- end }}
 {{- end }}
 {{- end }}
```
Comment (debugger24) [inline templates/server-job.yaml:303]:
```suggestion
{{- end }}
---
{{- end }}
```

File: values.yaml [hunk 4]
```diff
@@ -405,3 +405,10 @@ cassandra:
 
 mysql:
   enabled: false
+
+defaultnamespace:
+  setup:
+    enabled: false
+    name: default
+    backoffLimit: 100
+    retention: 1
```
Comment (fm2022aa) [inline values.yaml:414]: use bigger default value?

File: values.yaml [hunk 5]
```diff
@@ -405,3 +405,10 @@ cassandra:
 
 mysql:
   enabled: false
+
+defaultnamespace:
+  setup:
+    enabled: false
+    name: default
```
Comment (fm2022aa) [inline values.yaml:412]: can support multi name?

**Comment by robholland** [comment]
Implemented in https://github.com/temporalio/helm-charts/pull/550

**Pull request closed without merging**